src/algebra/lie_algebras.lean

```diff
@@ -0,0 +1,192 @@
+
+@[simp] lemma add_right (x y z : A) :
+  ⁅z, x + y⁆ = ⁅z, x⁆ + ⁅z, y⁆ :=
+by { unfold has_bracket.bracket, simp [right_distrib, left_distrib] }
+
+@[simp] lemma alternate (x : A) :
+  ⁅x, x⁆ = 0 :=
+by { unfold has_bracket.bracket, simp }
+
+lemma jacobi (x y z : A) :
+  ⁅x, ⁅y, z⁆⁆ + ⁅y, ⁅z, x⁆⁆ + ⁅z, ⁅x, y⁆⁆ = 0 :=
+begin
+  unfold has_bracket.bracket,
+  repeat { rw mul_sub_left_distrib },
+  repeat { rw mul_sub_right_distrib },
+  repeat { rw add_sub },
+  repeat { rw ←sub_add },
+  repeat { rw ←mul_assoc },
+  have h : ∀ (x y z : A), x - y + z + y = x+z := by simp,
+  repeat { rw h },
+  simp,
+end
+
+end ring_commutator
+
+/--
+A Lie ring is an additive group with compatible product, known as the bracket, satisfying the
+Jacobi identity. The bracket is not associative unless it is identically zero.
+-/
+class lie_ring (L : Type v) [add_comm_group L] extends has_bracket L :=
+(add_left : ∀ (x y z : L), ⁅x + y, z⁆ = ⁅x, z⁆ + ⁅y, z⁆)
+(add_right : ∀ (x y z : L), ⁅z, x + y⁆ = ⁅z, x⁆ + ⁅z, y⁆)
+(alternate : ∀ (x : L), ⁅x, x⁆ = 0)
+(jacobi : ∀ (x y z : L), ⁅x, ⁅y, z⁆⁆ + ⁅y, ⁅z, x⁆⁆ + ⁅z, ⁅x, y⁆⁆ = 0)
+
+namespace lie_ring
+
+variables (L : Type v) [add_comm_group L] [lie_ring L]
+
+attribute [simp] add_left
+attribute [simp] add_right
+attribute [simp] alternate
+attribute [simp] jacobi
+
+@[simp] protected lemma skew (x y : L) :
+  -⁅y, x⁆ = ⁅x, y⁆ :=
+begin
+  symmetry,
+  rw [←sub_eq_zero_iff_eq, sub_neg_eq_add],
+  have H : ⁅x + y, x + y⁆ = 0 := by apply lie_ring.alternate,
+  rw [lie_ring.add_left,
+      lie_ring.add_right, lie_ring.add_right,
+      lie_ring.alternate, lie_ring.alternate] at H,
+  simp at H,
+  exact H,
+end
+
+@[simp] protected lemma zero_left (x : L) :
+  ⁅x, 0⁆ = 0 :=
+begin
+  have H : ⁅x, 0⁆ + ⁅x, 0⁆ = ⁅x, 0⁆ + 0 := by { rw ←lie_ring.add_right, simp, },
+  exact add_left_cancel H,
+end
+
+@[simp] protected lemma zero_right (x : L) :
+  ⁅0, x⁆ = 0 := by { rw [←lie_ring.skew, lie_ring.zero_left], simp, }
+
+@[simp] protected lemma neg_left (x y : L) :
+  ⁅-x, y⁆ = -⁅x, y⁆ := by { rw [←sub_eq_zero_iff_eq, sub_neg_eq_add, ←lie_ring.add_left], simp, }
+
+@[simp] protected lemma neg_right (x y : L) :
+  ⁅x, -y⁆ = -⁅x, y⁆ := by { rw [←lie_ring.skew, ←lie_ring.skew], simp, }
+
+@[simp] protected lemma gsmul_left (x y : L) (n : ℤ) :
+  ⁅n • x, y⁆ = n • ⁅x, y⁆ :=
+begin
+  have H : ∀ (m : ℕ), ⁅m • x, y⁆ = m • ⁅x, y⁆ := by {
+    intros, induction m with m IHm,
+    { simp, },
+    { rw [←nat.one_add, add_smul, lie_ring.add_left, IHm, add_smul], simp, }, },
+  cases n,
+  { apply H, },
+  { unfold has_scalar.smul, unfold has_scalar.smul at H,
+    rw [gsmul_neg_succ, gsmul_neg_succ, lie_ring.neg_left, H], },
+end
+
+@[simp] protected lemma gsmul_right (x y : L) (n : ℤ) :
+  ⁅x, n • y⁆ = n • ⁅x, y⁆ :=
+begin
+  rw [←lie_ring.skew, ←lie_ring.skew _ x, lie_ring.gsmul_left],
+  unfold has_scalar.smul, rw gsmul_neg,
+end
+
+instance of_associative_ring (A : Type v) [ring A] : lie_ring A :=
```

Comment:
If the namespace is turned into a section, this would be `lie_ring.of_associative_ring`